test/e2e_node/swap_test.go

```diff
@@ -197,57 +406,137 @@ func expectLimitedSwap(f *framework.Framework, pod *v1.Pod, expectedSwapLimit in
-
 	return int64(swapAllocation)
 }
 
-func isLimitedSwap(f *framework.Framework, pod *v1.Pod) bool {
+func getSwapBehavior() string {
 	kubeletCfg, err := getCurrentKubeletConfig(context.Background())
 	framework.ExpectNoError(err, "cannot get kubelet config")
 
-	return kubeletCfg.MemorySwap.SwapBehavior == types.LimitedSwap
+	swapBehavior := kubeletCfg.MemorySwap.SwapBehavior
+	ginkgo.By("Figuring out swap behavior: " + swapBehavior)
+	return swapBehavior
 }
 
-func isNoSwap(f *framework.Framework, pod *v1.Pod) bool {
-	kubeletCfg, err := getCurrentKubeletConfig(context.Background())
-	framework.ExpectNoError(err, "cannot get kubelet config")
+func cloneQuantity(resource *resource.Quantity) *resource.Quantity {
+	clone := resource.DeepCopy()
+	return &clone
+}
+
+func divideQuantity(quantity *resource.Quantity, divideBy int64) *resource.Quantity {
+	dividedBigInt := new(big.Int).Div(
+		quantity.AsDec().UnscaledBig(),
+		big.NewInt(divideBy),
+	)
+
+	return resource.NewQuantity(dividedBigInt.Int64(), quantity.Format)
+}
+
+func multiplyQuantity(quantity *resource.Quantity, multiplier int64) *resource.Quantity {
+	product := new(big.Int).Mul(
+		quantity.AsDec().UnscaledBig(),
+		big.NewInt(multiplier),
+	)
+
+	return resource.NewQuantity(product.Int64(), quantity.Format)
+}
+
+func readCgroupFile(f *framework.Framework, pod *v1.Pod, cgroupFile string) (string, error) {
+	cgroupPath := filepath.Join(cgroupBasePath, cgroupFile)
+
+	outputStr := e2epod.ExecCommandInContainer(f, pod.Name, pod.Spec.Containers[0].Name, "sh", "-c", "cat "+cgroupPath)
+	outputStr = strings.TrimSpace(outputStr)
+	ginkgo.By("cgroup found value: " + outputStr)
+
+	return outputStr, nil
+}
 
-	return kubeletCfg.MemorySwap.SwapBehavior == types.NoSwap || kubeletCfg.MemorySwap.SwapBehavior == ""
+func parseBytesStrToQuantity(bytesStr string) (*resource.Quantity, error) {
+	bytesInt, err := strconv.ParseInt(bytesStr, 10, 64)
+	if err != nil {
+		return nil, fmt.Errorf("error parsing swap usage %s to int: %w", bytesStr, err)
+	}
+
+	return resource.NewQuantity(bytesInt, resource.BinarySI), nil
+}
+
+func getSwapUsage(f *framework.Framework, pod *v1.Pod) (*resource.Quantity, error) {
+	outputStr, err := readCgroupFile(f, pod, cgroupV2swapCurrentUsageFile)
+	if err != nil {
+		return nil, err
+	}
+
+	ginkgo.By("swap usage found: " + outputStr + " bytes")
+
+	return parseBytesStrToQuantity(outputStr)
+}
+
+func getMemoryUsage(f *framework.Framework, pod *v1.Pod) (*resource.Quantity, error) {
+	outputStr, err := readCgroupFile(f, pod, cgroupV2MemoryCurrentUsageFile)
+	if err != nil {
+		return nil, err
+	}
+
+	ginkgo.By("memory usage found: " + outputStr + " bytes")
+
+	return parseBytesStrToQuantity(outputStr)
+}
+
+// Sets memory request or limit can be null, then it's dismissed.
+// Sets the same value for all containers.
+func setPodMmoryResources(pod *v1.Pod, memoryRequest, memoryLimit *resource.Quantity) {
```

Comment:
is misspelling of the name intentional?

Reply:
It isn't, thanks for catching that! fixed